minions/async/src/main/java/io/github/microcks/minion/async/consumer/NATSMessageConsumptionTask.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to Laurent Broudoux (the "Author") under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Author licenses this
+ * file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.github.microcks.minion.async.consumer;
+
+import io.github.microcks.domain.Header;
+import io.github.microcks.minion.async.AsyncTestSpecification;
+import io.nats.client.Connection;
+import io.nats.client.Dispatcher;
+import io.nats.client.Nats;
+import io.nats.client.Options;
+import io.nats.client.impl.Headers;
+import org.jboss.logging.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An implementation of <code>MessageConsumptionTask</code> that consumes a
+ * topic on an NATS. Endpoint URL should be specified using the
+ * following form:
+ * <code>nats://{brokerhost[:port]}</code>
+ *
+ * @author laurent
+ */
+public class NATSMessageConsumptionTask implements MessageConsumptionTask {
+
+    /**
+     * Get a JBoss logging logger.
+     */
+    private final Logger logger = Logger.getLogger(getClass());
+
+    /**
+     * The string for Regular Expression that helps validating acceptable
+     * endpoints.
+     */
+    public static final String ENDPOINT_PATTERN_STRING = "nats://(?<brokerUrl>[^:]+(:\\d+)?)";
```

Comment:
I'm surprised you don't need a topic, queue, destination or subject name here so that you can later listen on specified subject ... ?

Reply:
I am definitely missing the connection of how consumption task can get the topic to listen to...

But where does that come from 🤔 I see the other protocols get this through the endpoint pattern string. Is that a microcks thing? Cause in general you cannot provide topic through the connection endpoint for NATS 🤔

Reply:
Yes, this a pure Microcks thing : it's up to you to decide which elements you'll need in an endpoint URL so that you'll be able to later use them for connecting the broker with required options in the consumption task. Sometimes, I get inspiration from how Camel endpoints are expressed, which options are provided, ... Check the [Camel for NATS endpoint](https://camel.apache.org/components/3.18.x/nats-component.html#_endpoint_options).